File: src/parallel_you/bus/message_broker.py
from parallel_you.model.bus import Transport, Message

class MessageBroker:
    def __init__(self, transport: Transport):
        self._transport = transport

    async def start(self) -> None:
        await self._transport.start()

    async def stop(self) -> None:
        await self._transport.stop()

    async def emit(self, type: str, payload: dict, meta: dict | None = None) -> None:
        msg = Message(type=type, payload=payload, meta=meta or {})
        await self._transport.publish(msg)

    async def subscribe(self, pattern: str, handler) -> None:
        await self._transport.subscribe(pattern, handler)

File: src/ParallelYou/ParallelYou.Bus/MessageBroker.cs
using ParallelYou.Bus.Abstractions;
using ParallelYou.Model.Messages;

namespace ParallelYou.Bus;

public class MessageBroker(ITransport transport) : IBroker
{
    public ITransport Transport { get; } = transport;
    
    public Task Publish(Message msg) => Transport.Publish(msg);

    public Task Emit(string routingKey, IReadOnlyDictionary<string, object> payload, IReadOnlyDictionary<string, object>? meta = null)
    {
        var msg = new SimpleMessage(payload, routingKey, meta);
        return Transport.Publish(msg);
    }

    public void Route(string routingKey, MessageHandler handler)
    {
        // fire-and-forget subscribe; tests can await Start()
        Transport.Subscribe(routingKey, handler).GetAwaiter().GetResult();
    }

    private sealed class SimpleMessage(
        IReadOnlyDictionary<string, object> payload,
        string? type,
        IReadOnlyDictionary<string, object>? meta)
        : Message<Dictionary<string, object>>(new Dictionary<string, object>(payload), type)
    {
        public IReadOnlyDictionary<string, object>? Meta { get; } = meta;
    }
}
